veps/sig-storage/incremental-backup.md

```diff
@@ -0,0 +1,374 @@
+          <source file='/run/kubevirt-private/vmi-disks/datavolumedisk/disk.img' index='3'/>
+        </dataStore>
+      </source>
+      ...
+    </disk>
+    <disk type='file' device='disk' model='virtio-non-transitional'>
+      <driver name='qemu' type='qcow2' cache='none' error_policy='stop' discard='unmap'/>
+      <source file='/run/kubevirt-private/libvirt/qemu/swtpm/datavolumedisk2.qcow2' index='4'>
+        <dataStore type='block'>
+          <format type='raw'/>
+          <source dev='/dev/datavolumedisk2' index='5'/>
+        </dataStore>
+      </source>
+      ...
+    </disk>
+```
+
+In the case the field is not present or set to `false`, the VM state PVC will deleted if it is not required by other features. The QCOW2 images will be deleted if exists. And the domain XML will be modified back to use the raw images as the disks images.
+
+
+### VirtualMachineBackupChain CR
+
+`VirtualMachineBackupChain` is a namespace-scoped Custom Resource designed to track backup checkpoints per-VM, per-backup-solution basis.
+It is intended to be created by a backup provider or user for a specific VM to monitor and manage incremental backups associated with a given backup solution. Upon creation, the `changedBlockTracking` field will be set to `true` if not already set.
+Checkpoint names within the backup chain will be prefixed with the backup chain's name, followed by a timestamp representing when the backup was taken.
+
+**Spec:**
+- `source` <br>
+Specifies the VM that this backup chain is associated with.
+- `checkpointsNames`<br>
+A list of checkpoint names managed by the backup controller.
+Once a corresponding checkpoint appears in the status.checkpoints list, it is considered valid and can be used as a base for subsequent incremental backups.
+If a checkpoint name is removed from checkpointsNames, it will trigger the deletion of the corresponding libvirt checkpoint. After successful deletion, the checkpoint will also be removed from status.checkpoints and will no longer be usable.
+
+**Status**
+- `latestCheckpoint` <br>
+The most recent checkpoint created, updated by the backup controller. This checkpoint will serve as the base for the next incremental backup.
+- `checkpoints`<br>
+A list of checkpoint entries. Each entry includes the checkpoint's name, creation timestamp, and metadata. Metadata is base64-encoded to prevent YAML clutter.
+
+* If the checkpoints list is empty, a full backup will be initiated.
+* If the list contains entries, an incremental backup will be performed using the `latestCheckpoint` as the base for determining which changed blocks to back up.
+
+*VirtualMachineBackupChain
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineBackupChain
+metadata:
+    name: my-backup-chain
+    namespace: ns1
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: my-vm
+    checkpointsNames:
+    - my-backup-chain-2025-03-03T16:13:28Z
+    - my-backup-chain-2025-03-03T17:13:28Z
```

Comment:
`checkpointNames` is a better name but why are these here in the first place? The spec should only be for things that the user can specify. Isn't `status.checkpoints` sufficient?